snowflake/repository.go

```diff
@@ -559,10 +559,25 @@ func (repo *SnowflakeRepository) GetColumnsInDatabase(databaseName string, handl
 }
 
 func (repo *SnowflakeRepository) CommentIfExists(comment, objectType, objectName string) error {
-	q := fmt.Sprintf(`COMMENT IF EXISTS ON %s %s IS '%s'`, objectType, objectName, comment)
-
+	q := fmt.Sprintf(`COMMENT IF EXISTS ON %s %s IS '%s'`, objectType, common.FormatQuery("%s", objectName), comment)
 	_, _, err := repo.query(q)
 
+	if err == nil {
+		return nil
+	}
+
+	if err != nil && objectType == "ROLE" {
+		logger.Warn("Encoutered error while commenting on role: " + err.Error())
+		ownershipQuery := common.FormatQuery(`GRANT OWNERSHIP ON ROLE %s TO ROLE %s`, objectName, repo.role)
+		_, _, err = repo.query(ownershipQuery)
+
+		if err != nil {
+			return err
```

Comment:
I would not return this error, but rather move the warning message here. 
If the initial error was not due to permissions, this would be a useless call.
In worst case, we just retry the original (set comment) action and return the error from that.